File: golang/12-god-object/account.go
package kata

type Account struct {
	Email          string
	Password       string
	Profile        map[string]string
	PaymentMethods []string
	Notifications  map[string]bool
	AuditLog       []string
	Subscription   string
}

func NewAccount(email, password string) *Account {
	return &Account{email, password, map[string]string{"name": "", "bio": ""}, []string{}, map[string]bool{"email": true, "sms": false}, []string{}, "basic"}
}
func (a *Account) login(pw string) bool {
	if pw == a.Password {
		a.AuditLog = append(a.AuditLog, "login")
		return true
	}
	return false
}
func (a *Account) logout() bool {
	a.AuditLog = append(a.AuditLog, "logout")
	return true
}
func (a *Account) update_profile(n, b string) map[string]string {
	a.Profile["name"] = n
	a.Profile["bio"] = b
	return a.Profile
}
func (a *Account) change_password(o, n string) bool {
	if o == a.Password {
		a.Password = n
		return true
	}
	return false
}
func (a *Account) add_payment_method(c string) int {
	a.PaymentMethods = append(a.PaymentMethods, c)
	return len(a.PaymentMethods)
}
func (a *Account) remove_payment_method(i int) []string {
	a.PaymentMethods = append(a.PaymentMethods[:i], a.PaymentMethods[i+1:]...)
	return a.PaymentMethods
}
func (a *Account) set_notification_preference(t string, v bool) map[string]bool {
	a.Notifications[t] = v
	return a.Notifications
}
func (a *Account) export_data() map[string]any {
	return map[string]any{
		"email":          a.Email,
		"profile":        a.Profile,
		"paymentMethods": a.PaymentMethods,
		"notifications":  a.Notifications,
		"auditLog":       a.AuditLog,
		"subscription":   a.Subscription,
	}
}
func (a *Account) log_access(action string) int {
	a.AuditLog = append(a.AuditLog, action)
	return len(a.AuditLog)
}
func (a *Account) check_subscription() string {
	return a.Subscription
}
func (a *Account) upgrade_subscription(p string) string { a.Subscription = p; return a.Subscription }

File: csharp/12-god-object/src/account.cs
public class Account
{
    public string Email;
    public string Password;

    public Dictionary<string, string> Profile = new()
    {
        { "name", "" },
        { "bio", "" }
    };

    public List<string> PaymentMethods = new();

    public Dictionary<string, bool> Notifications = new()
    {
        { "email", true },
        { "sms", false }
    };

    public List<string> AuditLog = new();
    public string Subscription = "basic";

    public Account(string e, string p)
    {
        Email = e;
        Password = p;
    }

    public bool login(string pw)
    {
        if (pw == Password)
        {
            AuditLog.Add("login:" + DateTimeOffset.UtcNow.ToUnixTimeMilliseconds());
            return true;
        }
        return false;
    }

    public Dictionary<string, string> update_profile(string n, string b)
    {
        Profile["name"] = n;
        Profile["bio"] = b;
        return Profile;
    }

    public bool change_password(string o, string n)
    {
        if (o == Password)
        {
            Password = n;
            return true;
        }
        return false;
    }

    public int add_payment_method(string c)
    {
        PaymentMethods.Add(c);
        return PaymentMethods.Count;
    }

    public List<string> remove_payment_method(int i)
    {
        PaymentMethods.RemoveAt(i);
        return PaymentMethods;
    }

    public Dictionary<string, bool> set_notification_preference(string t, bool v)
    {
        Notifications[t] = v;
        return Notifications;
    }

    public bool logout()
    {
        AuditLog.Add("logout:" + DateTimeOffset.UtcNow.ToUnixTimeMilliseconds());
        return true;
    }

    public dynamic export_data()
    {
        return new
        {
            email = Email,
            profile = Profile,
            paymentMethods = PaymentMethods,
            notifications = Notifications,
            auditLog = AuditLog,
            subscription = Subscription
        };
    }

    public int log_access(string action)
    {
        AuditLog.Add(action + ":" + DateTimeOffset.UtcNow.ToUnixTimeMilliseconds());
        return AuditLog.Count;
    }

    public string check_subscription()
    {
        return Subscription;
    }

    public string upgrade_subscription(string p)
    {
        Subscription = p;
        return Subscription;
    }
}
